src/adapters/quantcast.js

```diff
@@ -0,0 +1,135 @@
+const utils = require('../utils.js');
+const bidfactory = require('../bidfactory.js');
+const bidmanager = require('../bidmanager.js');
+const ajax = require('../ajax.js');
+const CONSTANTS = require('../constants.json');
+const QUANTCAST_CALLBACK_URL = 'http://global.qc.rtb.quantserve.com:8080/qchb';
+
+var QuantcastAdapter = function QuantcastAdapter() {
+
+  const BIDDER_CODE = 'quantcast';
+
+  const DEFAULT_BID_FLOOR = 0.0000000001;
+  let bidRequests = {};
+
+  let returnEmptyBid = function(bidId) {
+      var bidRequested = utils.getBidRequest(bidId);
+      if (!utils.isEmpty(bidRequested)) {
+        let bid = bidfactory.createBid(CONSTANTS.STATUS.NO_BID, bidRequested);
+        bid.bidderCode = BIDDER_CODE;
+        bidmanager.addBidResponse(bidRequested.placementCode, bid);
+      }
+      return;
+    };
+
+
+  //expose the callback to the global object:
+  $$PREBID_GLOBAL$$.handleQuantcastCB = function (responseText) {
+    if(utils.isEmpty(responseText)) {
+      return;
+    }
+    let response = null;
+    try {
+      response = JSON.parse(responseText);
+    } catch(e) {
+      // Malformed JSON
+      utils.logError("Malformed JSON received from server - can't do anything here");
+      return;
```

Comment:
@mkendall07 I tried to, but at that point in code we don't have the placement code for which the empty text response came and hence returning an error bid isn't possible..